docs/encyclopedia/temporal-nexus-service-operations.mdx

```diff
@@ -0,0 +1,146 @@
+   2. Retries are processed by the Nexus Machinery in the caller’s Temporal cluster and tracked as part of the caller Workflow, similar to Activities.
+3. Nexus `Operation handler` in the target cluster processes the Nexus `Operation`
+   1. `Operation handler` pulls operation `Task` from Task Queue
+   2. `Operation handler` starts a `Workflow` in the target cluster
+   3. Operation `result` is returned inline, since the operation is synchronous
+   4. **`NexusOperationCompleted`** added to caller’s event history
+4. Caller’s Workflow `handle.GetResult()` returns the result
+
+##### Executing Arbitrary (non-durable code) from a Sync Handler {#executing-arbitrary-(non-durable-code)-from-a-sync-handler}
+
+Nexus `Operation handlers` can execute arbitrary code, but unlike Activities they should be relatively short-lived, due to the lack of [heartbeats](https://community.temporal.io/t/best-practices-for-long-running-Activities/934) and a default 10 second NexusOperationStarted timeout. Nexus `Operation handlers` are not required to use underlying Temporal primitives like a `Workflow` or `Signal`.
+
+#### Automatic Retries
+
+Once the caller `Workflow` schedules an `Operation` with the caller’s Temporal cluster, the caller’s Nexus Machinery keeps trying to start the `Operation`, with automatic retries and exponential backoff. If a Nexus `Operation` returns a [retryable error](https://github.com/temporalio/temporal/blob/13d6cd8cf7a4ba0c4660cf98f672bbd645dca3e7/components/nexusoperations/executors.go#L659) when attempting to start, the `Operation` it will be retried up to the [default retry policy’s](https://github.com/temporalio/temporal/blob/de7c8879e103be666a7b067cc1b247f0ac63c25c/components/nexusoperations/config.go#L111) max attempts and expiration interval.
+
+#### Execution Semantics
+
+##### At-least-once Execution Semantics and Idempotency
+
+Since the caller's Nexus Machinery will keep trying to start the Operation over and over, the Nexus `Operation` handler should be idempotent just like Activities should be idempotent. It's not required in all cases, but highly recommended in general. With retries an Operation, like Activities, has at-least-once execution semantics, so an Operation should be idempotent as a general rule.
+
+##### At-most-once Execution Semantics via Underlying WorkflowIdReusePolicy
+
+To dedupe work and get at-most-once execution semantics, an `Operation` can start a `Workflow` with a `WorkflowIdReusePolicy` of `RejectDuplicates` which only allows one `Workflow Execution` per `Workflow Id` within a `Namespace` for the retention period.
+
+#### Versioning
+
+Task Routing is the simplest way to version your service code.
+
+If you have a new backward-incompatible Nexus `Operation` Handler, start by using a different Service and Task Queue. The version may be part of the service name, for example prod.payments.v2. Callers can then migrate to the new version in their normal deployment schedule.
```

Comment:
```suggestion
If you have a new backward-incompatible Nexus `Operation` Handler, start by using a different Service and Task Queue. 
The version may be part of the service name, for example prod.payments.v2. 
Callers can then migrate to the new version in their normal deployment schedule.
```

Reply:
I'd probably recommend a separate endpoint TBH.
There's more to say on this topic, wire level compatibility is a topic that should be familiar to most devs but should be mentioned here for completeness.

Reply:
Why different endpoint? To avoid versioning in your service name? Seems like a separate endpoint would be a 3rd option that may be useful, but would require more config (copying endpoint allowlist, etc.) and for all clients to update to use the new endpoint name and you have to maintain 2 worker fleets while everyone is migrating. This might be ok now with a single upstream target, but as endpoints get scoped more broadly (like prod-east or prod-west) it may be much harder to spin up a new endpoint just because one op had a breaking change.
```suggestion
If you have a new backward-incompatible Nexus `Operation` Handler, for example due to a wire-level incompatible breaking change, start by using a different Service and Task Queue. The version may be part of the service name, for example `prod.payments.v2`. Callers can then migrate to the new version in their normal deployment schedule.
```